libnetwork/drivers/bridge/setup_ip_tables_linux_test.go

```diff
@@ -499,3 +482,87 @@ func TestMirroredWSL2Workaround(t *testing.T) {
 		})
 	}
 }
+
+func setupWSL2MirroredMode(t *testing.T, loopback0 bool, wslinfoPerm os.FileMode) func() {
+	if loopback0 {
+		iface := &netlink.Dummy{
+			LinkAttrs: netlink.LinkAttrs{
+				Name: "loopback0",
+			},
+		}
+		err := netlink.LinkAdd(iface)
+		assert.NilError(t, err)
+	}
+
+	wslinfoPathOrig := wslinfoPath
+	if wslinfoPerm != 0 {
+		tmpdir := t.TempDir()
+		wslinfoPath = filepath.Join(tmpdir, "wslinfo")
+		err := os.WriteFile(wslinfoPath, []byte("#!/bin/sh\necho dummy file\n"), wslinfoPerm)
+		assert.NilError(t, err)
+	}
+
+	return func() {
+		wslinfoPath = wslinfoPathOrig
+	}
+}
+
+func TestMirroredWSL2LoopbackFiltering(t *testing.T) {
+	for _, tc := range []struct {
+		desc             string
+		loopback0        bool
+		wslinfoPerm      os.FileMode // 0 for no-file
+		expLoopback0Rule bool
+	}{
+		{
+			desc: "No loopback0",
+		},
+		{
+			desc:             "WSL2 mirrored",
+			loopback0:        true,
+			wslinfoPerm:      0o777,
+			expLoopback0Rule: true,
+		},
+		{
+			desc:        "loopback0 but wslinfo not executable",
+			loopback0:   true,
+			wslinfoPerm: 0o666,
+		},
+		{
+			desc:      "loopback0 but no wslinfo",
+			loopback0: true,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			defer netnsutils.SetupTestOSContext(t)()
+			defer setupWSL2MirroredMode(t, tc.loopback0, tc.wslinfoPerm)()
+
+			nw := bridgeNetwork{
+				driver: &driver{
+					config: configuration{EnableIPTables: true},
+				},
+			}
+			err := nw.filterPortMappedOnLoopback(portBinding{
+				PortBinding: types.PortBinding{
+					Proto:    types.TCP,
+					IP:       net.ParseIP("127.0.0.1"),
+					HostPort: 8000,
+				},
+				childHostIP: net.ParseIP("127.0.0.1"),
+			}, true)
+			assert.NilError(t, err)
+
+			out, err := exec.Command("iptables-save", "-t", "raw").CombinedOutput()
+			assert.NilError(t, err)
+
+			if tc.expLoopback0Rule {
+				assert.Assert(t, strings.Count(string(out), "-A PREROUTING") == 2)
+				assert.Assert(t, is.Contains(string(out), "-A PREROUTING -d 127.0.0.1/32 -i loopback0 -p tcp -m tcp --dport 8000 -j ACCEPT"))
+			} else {
+				assert.Assert(t, strings.Count(string(out), "-A PREROUTING") == 1)
+				// There should be no rule in the raw-PREROUTING chain.
+				assert.Assert(t, !strings.Contains(string(out), "loopback0"))
```

Comment:
Same here (I wish there was a better way to handle the "not contains" cases); also, might as well put the comment in the "custom comment" of the assert, so that it's logged on failure;

```go
assert.Check(t, is.Equal(strings.Count(string(out), "-A PREROUTING"), 1))
assert.Check(t, !strings.Contains(string(out), "loopback0"), "There should be no rule in the raw-PREROUTING chain")
```